js/classes/entities/midground/enemies/ground_enemy_base.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * Class which provides basic functionality for enemy entities that move on the ground.
+ * Handles movement, block collisions, etc.
+ * 
+ * @author Trae Claar
+ */
+class GroundEnemyBase {
+
+    /**
+     * Constructor for a GroundEnemyBase.
+     * 
+     * @param {Entity} enemy the enemy associated with this EnemyBase
+     * @param {Vector} pos the initial position of the enemy
+     * @param {number} scaledSize the size of the enemy
+     * @param {number} speed the speed of the enemy
+     * @param {number} health the initial health of the enemy
+     * @param {number} maxRoamDistance the maximum distance the enemy will wander 
+     *      from its initial position when roaming
+     * @param {function} onDeath callback function invoked when the enemy dies
+     * @param {string} stance the enemy's stance (use an EnemyBase member stance, 
+     *      e.g. EnemyBase.AggressiveStance)
+     * @param {number} reactionDistance the maximum distance that the enemy will 
+     *      react to Chad according to its stance and state
+     */
+    constructor(enemy, pos, scaledSize, speed, health, maxRoamDistance, onDeath, stance, reactionDistance) {
+        this.enemy = enemy;
+        this.base = new EnemyBase(enemy, pos, scaledSize, speed, health, onDeath, () => this.handleDamage());
+        
+        enemy.state = "roam";
+        enemy.action = "idle";
+        enemy.maxRoamDistance = maxRoamDistance; // *
+        enemy.yVelocity = 0;
+
+        enemy.getDirection = () => this.getDirection();
+
+        this.minRoamX = pos.x - maxRoamDistance;
+        this.targetX = pos.x;
+        this.stance = stance;
+        this.reactionDistance = reactionDistance ?? Camera.SIZE.x / 2 - scaledSize.x;
```

Comment:
Didn't know that ?? is an operator. Cool! Looks like it checks for undefined values and handles that from what I'm learning online.

Reply:
Yes, it returns the left operand if it is not nullish (null or undefined), and the right operand otherwise.